frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:appflowy/startup/plugin/plugin.dart';
+import 'package:appflowy/workspace/application/view/view_ext.dart';
+import 'package:appflowy/workspace/application/view/view_listener.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder2/view.pb.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class DatabaseViewWidget extends StatefulWidget {
+  const DatabaseViewWidget({required this.view, super.key});
+
+  final ViewPB view;
+
+  @override
+  State<DatabaseViewWidget> createState() => _DatabaseViewWidgetState();
+}
+
+class _DatabaseViewWidgetState extends State<DatabaseViewWidget> {
+  /// Listens to the view updates.
+  late ViewListener _listener;
+
+  /// Notifies the view layout type changes. When the layout type changes,
+  /// the widget of the view will be updated.
+  late ViewLayoutTypeChangeNotifier _layoutTypeChangeNotifier;
+
+  /// The view will be updated by the [ViewListener].
+  late ViewPB view;
+
+  @override
+  void initState() {
+    view = widget.view;
+    _listener = ViewListener(view: widget.view);
+    _listener.start(
+      onViewUpdated: (result) {
+        result.fold(
+          (updatedView) {
+            if (mounted) {
+              view = updatedView;
+              _layoutTypeChangeNotifier.setLayoutType(view.layout);
+            }
+          },
+          (r) => null,
+        );
+      },
+    );
+
+    _layoutTypeChangeNotifier = ViewLayoutTypeChangeNotifier(
+      layoutType: widget.view.layout,
+    );
+
+    super.initState();
```

Comment:
Best practice to have the super.initState() first.

Reply:
Got it.